estate/models/estate_property.py

```diff
@@ -0,0 +1,120 @@
+    selling_price = fields.Float(string='Selling Price', readonly=True, copy=False)
+    best_price = fields.Float(string='Best Price', compute='_compute_best_price', store=True)
+    property_type_id = fields.Many2one('estate.property.type', string='Property Type')
+    tag_ids = fields.Many2many('estate.property.tag', string='Tags')
+    buyer_id = fields.Many2one('res.partner', string='Buyer')
+    seller_id = fields.Many2one('res.users', string='Seller', default=lambda self: self.env.user)
+    bedrooms = fields.Integer(string='Bedrooms', default=2)
+    living_area = fields.Integer(string='Living Area (sqm)')
+    facades = fields.Integer(string='Facades')
+    garage = fields.Boolean(string='Garage')
+    garden = fields.Boolean(string='Garden')
+    garden_area = fields.Integer(string='Garden Area')
+    garden_orientation = fields.Selection(
+        [('north', 'North'), ('south', 'South'), ('east', 'East'), ('west', 'West')],
+        string='Garden Orientation',
+    )
+    total_area = fields.Integer(string='Total Area (sqm)', compute='_compute_total_area', store=True)
+    state = fields.Selection(
+        [
+            ('new', 'New'),
+            ('offer_received', 'Offer Received'),
+            ('offer_accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled'),
+        ],
+        string='State',
+        default='new',
+        required=True,
+        copy=False,
+        compute='_compute_state',
+        store=True,
+    )
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string='Offers')
+    active = fields.Boolean(string='Active', default=True)
+
+    ###### COMPUTE ######
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends('offer_ids.price')
+    def _compute_best_price(self):
+        for record in self:
+            prices = record.offer_ids.mapped('price')
+            if prices:
+                record.best_price = max(prices)
+            else:
+                record.best_price = 0.0
+
+    @api.depends('offer_ids')
+    def _compute_state(self):
+        for record in self:
+            if not record.offer_ids:
+                record.state = 'new'
+            elif record.state == 'new':
+                record.state = 'offer_received'
+
+    ###### ONCHANGE ######
+    @api.onchange('garden')
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = 0
+            self.garden_orientation = False
+
+    ###### CRUD ######
+    def unlink(self):
+        if any(record.state not in ['new', 'cancelled'] for record in self):
+            raise UserError('You cannot delete a property that is not new or cancelled.')
```

Comment:
```suggestion
            raise UserError(self.env._('You cannot delete a property that is not new or cancelled.'))
```
As Séna would say:

> In Odoo, translatable strings in "data"-type content are automatically exported for translation. However, other user-facing strings, Odoo cannot automatically know whether they are translatable. To ensure these strings are exported for translation, you need to explicitly mark them. This is achieved by wrapping the literal string with the self.env._() function. The _() function signals to Odoo's translation system that the string should be included in the translation catalog.
> 
> More info here: https://www.odoo.com/documentation/18.0/developer/howtos/translations.html#explicit-exports
> 

You can check this line here:
https://github.com/odoo/odoo/blob/05cff3b7d866f6bc95c4b32f343ae14a4da946f2/addons/account/models/account_move.py#L2075